.github/workflows/e2e-tests.yml

```diff
@@ -95,12 +97,17 @@ jobs:
         ./bin/env.sh start
 
     - name: Run all tests
-      if: matrix.gutenberg == 'no'
+      if: github.event_name != 'schedule' && matrix.gutenberg == 'no'
       working-directory: projects/plugins/jetpack/tests/e2e
       run: yarn test-e2e
 
     - name: Run only blocks tests with latest Gutenberg
-      if: matrix.gutenberg == 'latest'
+      if: github.event_name != 'schedule' && matrix.gutenberg == 'latest'
+      working-directory: projects/plugins/jetpack/tests/e2e
+      run: yarn test-e2e --testNamePattern=blocks
+
+    - name: Scheduled run blocks tests with pre-release Gutenberg
```

Comment:
This approach is kinda messy. We would end up with 3 runs for both `push` and `schedule` events. All of the "skipped" jobs would build Jetpack and upload artifacts anyway.

Reply:
You could do like I did for tests.yml: One lightweight job that produces the matrix, then the matrix job depends on that to use the generated matrix. Too bad GH doesn't have any other way to DRY when you need all the same setup in several similar but not-quite-identical jobs.

Even better would be if the E2E setup and running could be closer to how the rest of the tests do it so we could run them from the same matrix. But that's probably longer term than this PR, as it would probably require dropping the use of wp-env.

Reply:
I wonder if we really need to no/latest variation on every PR. With this new pre-release maybe we can only run the GB != bundled in a schedule, and have only a full suite run with the bundled GB in PRs checks. We could then have 2 different workflows. 
* Optional we could make the PR workflow depend on the build one, and use the build artefacts as input, so we don't build twice. This will also remove all the PHP setup as we don't need it to run the E2E tests. 
* The second workflow will be similar to the one we have now, with a matrix and a Jetpack build, but no event condition.

Reply:
How about collapsing these two steps into one, and merging their `if` expressions into one?

```
    - name: Run blocks tests with ${{ matrix.gutenberg }} Gutenberg
      if: |
        ( github.event_name != 'schedule' && matrix.gutenberg == 'latest' ) ||
        ( github.event_name == 'schedule' && matrix.gutenberg == 'pre-release' )
      working-directory: projects/plugins/jetpack/tests/e2e
      run: yarn test-e2e --testNamePattern=blocks
```

> This approach is kinda messy. We would end up with 3 runs for both `push` and `schedule` events. All of the "skipped" jobs would build Jetpack and upload artifacts anyway.

You could build Jetpack in a separate job (without a matrix) and upload the build as an artifact, declare that the test job `needs:` the build job, and download the artifact there. (We have some [precedent](https://github.com/WordPress/gutenberg/blob/2f1b4dc85b0617d60c1da0e215e8ed2da6eb2cea/.github/workflows/build-plugin-zip.yml) for something like that in the Gutenberg repo: See how the `create-release` job uses the artifact from the `build` job.)

As for the uploading of the test result artifacts, you could try changing the step's `if:` from `${{ always() }}` to `success() || failure()` (to maybe skip it if the previous step is skipped?) Not sure if that's going to work, though.

As a further/later optimization, it's probably even possible to also absorb the "run all tests" step into this one (e.g. through the use of an env variable).

Reply:
> I wonder if we really need to run no/latest variation on every PR. With this new pre-release maybe we can only run the GB != bundled in a schedule, and have only a full suite run with the bundled GB in PRs checks.

Yeah, I don't think we really need to run `latest` GB tests on every PR. We probably can get rid of the matrix in the job altogether. 

Moving the build step into a separate job is a good idea. I'll look into it in a follow up PR

Reply:
with the latest changes from @adimoldovan around E2E workflow, I made the matrix defined in runtime. During the scheduled run, an additional job would be triggered that would run block tests with the latest GB plugin.